Adds a flag to allow showing the source of the message

donatj opened the pull request:
With Phan (and likely others) it's nice to know what specific rule you're breaking. 

This simply adds a flag `--show-source` that prefixes the message with the source attribute of the error line.

```xml
  <file name="application/routes/www/user/user.php">
    <error line="73" severity="info" message="Code $cdnPath has a dynamic format string that could not be inferred by Phan" source="PhanPluginPrintfVariableFormatString"/>
  </file>
```

So this goes from `Code $cdnPath has a dynamic format string that could not be inferred by Phan`

to: `PhanPluginPrintfVariableFormatString: Code $cdnPath has a dynamic format string that could not be inferred by Phan`

staabm commented on cs2pr:
```diff
@@ -31,6 +32,9 @@ foreach ($argv as $arg) {
         case 'graceful-warnings':
             $gracefulWarnings = true;
             break;
+        case 'show-source':
```
Thx for the PR.

I feel this option should be named `prepend-source` or similar.
Additionally the new option should be added to the help output.

staabm commented on cs2pr:
```diff
@@ -87,6 +91,10 @@ foreach ($root as $file) {
         $message = (string) $error['message'];
 
+        if($showSource) {
+            $message = ((string) $error['source']).': '.$message;
```
This requires a array_key_exists() or similar check beforehand, as not every error will have a „source“?

donatj commented:
Alrighty, fixed this one up.

Changed it to only show the source if there _is_ a source attribute which has cleaner output on mixed sets. I'm not sure if mixed sets pop up in nature, but who knows.

staabm reviewed (approved):
Just dropped a debug leftover - but except that, this one looks perfect. Thanks for the contribution <3

donatj commented:
Oops, sorry about the require-dev. I genuinely don't know how I committed that. ;)
